src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -828,14 +832,15 @@ struct BlockZipperJoinImpl {
   // `rightSide_.currentBlocks`) s.t. only elements `> lastProcessedElement`
   // remain. This effectively removes all blocks completely, except maybe the
   // last one.
-  template <typename Blocks, typename ProjectedEl>
-  void removeEqualToCurrentEl(Blocks& blocks,
-                              ProjectedEl lastProcessedElement) {
+  void removeEqualToCurrentEl(
+      ad_utility::SameAsAny<typename LeftSide::CurrentBlocks,
+                            typename RightSide::CurrentBlocks> auto& blocks,
+      const ProjectedEl& lastProcessedElement) {
     // Erase all but the last block.
     AD_CORRECTNESS_CHECK(!blocks.empty());
     if (blocks.size() > 1 && !blocks.front().empty()) {
-      AD_CORRECTNESS_CHECK(!lessThan_(lastProcessedElement,
-                                      std::as_const(blocks.front()).back()));
+      AD_CORRECTNESS_CHECK(
+          !lessThan_(lastProcessedElement, blocks.front().back()));
```

Comment:
Here `as_const` is dropped. This presumably works because of recent fixes to the `IdTable` class.